app/src/ui/branches/pull-request-list-item.tsx

```diff
@@ -51,7 +51,11 @@ export class PullRequestListItem extends React.Component<
   public render() {
     const title = this.props.loading === true ? undefined : this.props.title
     const subtitle = this.getSubtitle()
-
+    const matches = this.props.matches
+    const hasTitleMatches = matches.length > 0
+    const titleMatches = hasTitleMatches ? matches[0] : []
+    const hasSubtitleMatches = matches.length > 1
+    const subtitleMatches = hasSubtitleMatches ? matches[1] : []
```

Comment:
I think at this point the only concern that I have with this change is that the `matches` result is now an array of arrays, and we need to carry around the implicit context like this.

It'd be nice to move this to a more declarative shape, like this:

```ts
interface IMatches {
  readonly title: ReadonlyArray<number>,
  readonly subtitle: ReadonlyArray<number>
}
```

But I think that means revisiting `IMatch<T>` because that's where we "carry around" these maches:

```ts
export interface IMatch<T> {
  /** `0 <= score <= 1` */
  score: number
  item: T
  matches: ReadonlyArray<ReadonlyArray<number>>
}
```

I won't hold up this PR for that, but if anyone has bright ideas I'd love to hear them!

Reply:
I think I like your suggestion. Let me try it out